lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -49,6 +50,136 @@ export class TokenValidator {
         this.openIdConfigProvider = new OpenIdConfigProvider(this.config, this.networkInterface, this.logger);
     }
 
+    /**
+     * Middleware-style function to validate token from request.
+     * @param {@link (TokenValidationParameters:type)}
+     * @param resource Optional resource to retrieve access token from session 
+     * @returns 
+     */
+    validateTokenMiddleware(options: TokenValidationParameters, resource?: string): Function {
```

Comment:
Instead of just `Function`, can we set a type with the expected signature?

Reply:
Makes sense. Added response type.